### PR TITLE
Implement `with_row_count` for `DataFrame` and `LazyFrame`

### PR DESCRIPTION
**Pull request by Sicheng-Pan**
#236

### CODE REVIEW

**Review by sorhawell — APPROVED** [review]
Many thanks ! :)

**Comment by eitsupi** [comment]
@sorhawell Why are you merging this before #308?

File: R/extendr-wrappers.R [hunk 1]
```diff
@@ -1,7 +1,4 @@
 # Generated by extendr: Do not edit by hand
-
-# nolint start
-
```
Comment (eitsupi) [inline R/extendr-wrappers.R:4]: Please use the latest rextendr

Comment (Sicheng-Pan) [reply R/extendr-wrappers.R:4]: I tried `rextendr-0.3.1`, but this comments seems to show up as well. Do you mean the latest commit of `rextendr`?

Comment (eitsupi) [reply R/extendr-wrappers.R:4]: The addition of this comment is intentional (rextendr >= 0.3.0); see extendr/rextendr#214.

**Pull request merged**